diff --git a/src/util/sys_defs.h b/src/util/sys_defs.h
index 2f0acb9..448a309 100644
--- a/src/util/sys_defs.h
+++ b/src/util/sys_defs.h
@@ -709,7 +709,7 @@ extern int initgroups(const char *, int);
  /*
   * LINUX.
   */
-#if defined(LINUX2) || defined(LINUX3)
+#if defined(LINUX2) || defined(LINUX3) || defined(LINUX4) || defined(LINUX5)
 #define SUPPORTED
 #include <sys/types.h>
 #define UINT32_TYPE	unsigned int
