diff --git a/tests/dateformat_format_50+.phpt b/tests/dateformat_format_50+.phpt
index 92966c3..a39b373 100644
--- a/tests/dateformat_format_50+.phpt
+++ b/tests/dateformat_format_50+.phpt
@@ -323,7 +323,7 @@ IntlDateFormatter locale= en_US ,datetype = -1 ,timetype =-1
 Formatted localtime_array is : 18951217 12:13 AM
 ------------
 Date is: DateTime::__set_state(array(
-   'date' => '2010-01-01 01:02:03',
+   'date' => '2010-01-01 01:02:03.000000',
    'timezone_type' => 3,
    'timezone' => 'UTC',
 ))
@@ -331,7 +331,7 @@ Date is: DateTime::__set_state(array(
 Formatted DateTime is : Thursday, December 31, 2009 at 3:02:03 PM GMT-10:00
 ------------
 Date is: DateTime::__set_state(array(
-   'date' => '2010-01-01 01:02:03',
+   'date' => '2010-01-01 01:02:03.000000',
    'timezone_type' => 3,
    'timezone' => 'UTC',
 ))
@@ -339,7 +339,7 @@ Date is: DateTime::__set_state(array(
 Formatted DateTime is : December 31, 2009 at 3:02:03 PM GMT-10
 ------------
 Date is: DateTime::__set_state(array(
-   'date' => '2010-01-01 01:02:03',
+   'date' => '2010-01-01 01:02:03.000000',
    'timezone_type' => 3,
    'timezone' => 'UTC',
 ))
@@ -347,7 +347,7 @@ Date is: DateTime::__set_state(array(
 Formatted DateTime is : Dec 31, 2009, 3:02:03 PM
 ------------
 Date is: DateTime::__set_state(array(
-   'date' => '2010-01-01 01:02:03',
+   'date' => '2010-01-01 01:02:03.000000',
    'timezone_type' => 3,
    'timezone' => 'UTC',
 ))
@@ -355,7 +355,7 @@ Date is: DateTime::__set_state(array(
 Formatted DateTime is : 12/31/09, 3:02 PM
 ------------
 Date is: DateTime::__set_state(array(
-   'date' => '2010-01-01 01:02:03',
+   'date' => '2010-01-01 01:02:03.000000',
    'timezone_type' => 3,
    'timezone' => 'UTC',
 ))
@@ -363,7 +363,7 @@ Date is: DateTime::__set_state(array(
 Formatted DateTime is : 20091231 03:02 PM
 ------------
 Date is: DateTime::__set_state(array(
-   'date' => '2000-12-30 19:04:05',
+   'date' => '2000-12-30 19:04:05.000000',
    'timezone_type' => 3,
    'timezone' => 'America/Los_Angeles',
 ))
@@ -371,7 +371,7 @@ Date is: DateTime::__set_state(array(
 Formatted DateTime is : Saturday, December 30, 2000 at 5:04:05 PM GMT-10:00
 ------------
 Date is: DateTime::__set_state(array(
-   'date' => '2000-12-30 19:04:05',
+   'date' => '2000-12-30 19:04:05.000000',
    'timezone_type' => 3,
    'timezone' => 'America/Los_Angeles',
 ))
@@ -379,7 +379,7 @@ Date is: DateTime::__set_state(array(
 Formatted DateTime is : December 30, 2000 at 5:04:05 PM GMT-10
 ------------
 Date is: DateTime::__set_state(array(
-   'date' => '2000-12-30 19:04:05',
+   'date' => '2000-12-30 19:04:05.000000',
    'timezone_type' => 3,
    'timezone' => 'America/Los_Angeles',
 ))
@@ -387,7 +387,7 @@ Date is: DateTime::__set_state(array(
 Formatted DateTime is : Dec 30, 2000, 5:04:05 PM
 ------------
 Date is: DateTime::__set_state(array(
-   'date' => '2000-12-30 19:04:05',
+   'date' => '2000-12-30 19:04:05.000000',
    'timezone_type' => 3,
    'timezone' => 'America/Los_Angeles',
 ))
@@ -395,7 +395,7 @@ Date is: DateTime::__set_state(array(
 Formatted DateTime is : 12/30/00, 5:04 PM
 ------------
 Date is: DateTime::__set_state(array(
-   'date' => '2000-12-30 19:04:05',
+   'date' => '2000-12-30 19:04:05.000000',
    'timezone_type' => 3,
    'timezone' => 'America/Los_Angeles',
 ))
