diff --git a/configure b/configure
index 7e6e33a..ad1c10f 100755
--- a/configure
+++ b/configure
@@ -16,7 +16,7 @@ test -d $NGX_OBJS || mkdir -p $NGX_OBJS
 echo > $NGX_AUTO_HEADERS_H
 echo > $NGX_AUTOCONF_ERR
 
-echo "#define NGX_CONFIGURE \"$NGX_CONFIGURE\"" > $NGX_AUTO_CONFIG_H
+echo "#define NGX_CONFIGURE \"<omitted, take a look at nix expression>\"" > $NGX_AUTO_CONFIG_H
 
 
 if [ $NGX_DEBUG = YES ]; then
