--- a/configure.in	2020-11-20 15:31:41.358413994 +0300
+++ b/configure.in	2020-11-20 15:31:54.158612427 +0300
@@ -195,15 +195,6 @@
 AX_CODE_COVERAGE
 
 AC_CONFIG_FILES([Makefile
-	tests/Makefile
-	tests/ftests/Makefile
-	tests/gunit/Makefile
-	tests/tools/testenv.sh
-	tests/tools/Makefile
-	tests/tools/cgconfigparser/Makefile
-	tests/tools/cgclassify/Makefile
-	tests/tools/multimount/Makefile
-	tests/runlibcgrouptest.sh
 	src/Makefile
 	src/daemon/Makefile
 	src/tools/Makefile
--- a/src/Makefile.am	2020-11-20 15:34:52.597378619 +0300
+++ b/src/Makefile.am	2020-11-20 15:41:15.999309734 +0300
@@ -14,15 +14,9 @@
 CLEANFILES = lex.c parse.c parse.h
 
 INCLUDES = -I$(top_srcdir)/include
-lib_LTLIBRARIES = libcgroup.la libcgroupfortesting.la
+lib_LTLIBRARIES = libcgroup.la
 libcgroup_la_SOURCES = parse.h parse.y lex.l api.c config.c libcgroup-internal.h libcgroup.map wrapper.c log.c
 libcgroup_la_LIBADD = -lpthread $(CODE_COVERAGE_LIBS)
 libcgroup_la_CFLAGS = $(CODE_COVERAGE_CFLAGS) -DSTATIC=static
 libcgroup_la_LDFLAGS = -Wl,--version-script,$(srcdir)/libcgroup.map \
 	-version-number $(LIBRARY_VERSION_MAJOR):$(LIBRARY_VERSION_MINOR):$(LIBRARY_VERSION_RELEASE)
-
-libcgroupfortesting_la_SOURCES = $(libcgroup_la_SOURCES)
-libcgroupfortesting_la_LIBADD = -lpthread $(CODE_COVERAGE_LIBS)
-libcgroupfortesting_la_CFLAGS = $(CODE_COVERAGE_CFLAGS) -DSTATIC= -DUNIT_TEST
-libcgroupfortesting_la_LDFLAGS = -Wl,--version-script,$(top_srcdir)/tests/gunit/libcgroup_unittest.map \
-	-version-number $(LIBRARY_VERSION_MAJOR):$(LIBRARY_VERSION_MINOR):$(LIBRARY_VERSION_RELEASE)
--- a/Makefile.am	2020-11-20 15:43:07.433029682 +0300
+++ b/Makefile.am	2020-11-20 15:43:11.053085558 +0300
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = foreign
 ACLOCAL_AMFLAGS= -I m4
-SUBDIRS = dist doc include samples scripts src tests
+SUBDIRS = dist doc include samples scripts src
 
 EXTRA_DIST = README_daemon libcgroup.doxyfile README_systemd
 
--- a/src/tools/Makefile.am	2020-11-20 15:44:59.754763433 +0300
+++ b/src/tools/Makefile.am	2020-11-20 15:45:05.486851914 +0300
@@ -54,7 +54,4 @@
 cgsnapshot_LIBS = $(CODE_COVERAGE_LIBS)
 cgsnapshot_CFLAGS = $(CODE_COVERAGE_CFLAGS)
 
-install-exec-hook:
-	chmod u+s $(DESTDIR)$(bindir)/cgexec
-
 endif
