--- a/Makefile.am	2020-11-20 15:43:07.433029682 +0300
+++ b/Makefile.am	2020-11-20 15:43:11.053085558 +0300
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = foreign
 ACLOCAL_AMFLAGS= -I m4
-SUBDIRS = dist doc include samples scripts src tests
+SUBDIRS = dist doc include samples scripts src
 
 EXTRA_DIST = README_daemon libcgroup.doxyfile README_systemd
 
--- a/src/tools/Makefile.am	2020-11-20 15:44:59.754763433 +0300
+++ b/src/tools/Makefile.am	2020-11-20 15:45:05.486851914 +0300
@@ -54,7 +54,4 @@
 cgsnapshot_LIBS = $(CODE_COVERAGE_LIBS)
 cgsnapshot_CFLAGS = $(CODE_COVERAGE_CFLAGS)
 
-install-exec-hook:
-	chmod u+s $(DESTDIR)$(bindir)/cgexec
-
 endif
