--- a/src/cron.c	2018-11-26 21:08:16.000000000 +0300
+++ b/src/cron.c	2019-10-29 19:00:59.814833352 +0300
@@ -23,6 +23,7 @@
 /* Forward declarations.  */
 static void inner_main (void *closure, int argc, char *argv[]);
 static void react_to_terminal_signal (int sig);
+static void react_to_hup_signal (int sig);
 static SCM set_cron_signals (void);
 
 int
@@ -64,11 +65,19 @@
   sigaction (SIGTERM, &sa, 0);
   sigaction (SIGINT,  &sa, 0);
   sigaction (SIGQUIT, &sa, 0);
+
+  sa.sa_handler = react_to_hup_signal;
   sigaction (SIGHUP,  &sa, 0);
 
   return SCM_BOOL_T;
 }
 
+static void
+react_to_hup_signal (int sig)
+{
+  scm_c_eval_string ("(%process-files #f #t)");
+}
+
 /* Handle signal SIG and exit.  All signals that cron handles will produce the
    same behavior so we don't need to use SIG in the implementation.  */
 static void
