--- a/src/mcron/scripts/crontab.scm	1970-01-01 03:00:01.000000000 +0300
+++ b/src/mcron/scripts/crontab.scm	2019-10-30 17:58:12.680215035 +0300
@@ -215,8 +215,7 @@
                    (with-output-to-file crontab-file
                      (λ () (display input-string))))
                  (begin
-                   (read-vixie-file input-file)
-                   (copy-file input-file crontab-file))))
+                   (read-vixie-file input-file))))
             (hit-server crontab-user)))
 
          ;; The user is being silly. The message here is identical to the one
