diff --git a/src/mcron/scripts/cron.scm b/src/mcron/scripts/cron.scm
index 1a97fdf..4563d7e 100644
--- a/src/mcron/scripts/cron.scm
+++ b/src/mcron/scripts/cron.scm
@@ -173,6 +173,7 @@ option.\n")
       ;; process the socket request before restarting the loop again.
       (catch-mcron-error
        (let ((fdes-list (cron-file-descriptors)))
+         (call-with-new-thread (lambda () (while #t (begin (child-cleanup) (sleep 2)))))
          (while #t
            (run-job-loop fdes-list)
            (unless (null? fdes-list)
