From 360212149cdf3baa284f7fc7a5f1712c9c6c30b0 Mon Sep 17 00:00:00 2001
From: Oleg Pykhalov <go.wigust@gmail.com>
Date: Tue, 3 Sep 2019 17:55:57 +0300
Subject: [PATCH] mcron-core.scm: Display job to STDOUT.

---
 mcron-core.scm | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mcron-core.scm b/mcron-core.scm
index 518bcac..10239ac 100644
--- a/mcron-core.scm
+++ b/mcron-core.scm
@@ -198,6 +198,7 @@
   (for-each (lambda (job)
               (if (eqv? (primitive-fork) 0)
                   (begin
+                    (format #t "CMD (~a)" (job:action job))
                     (setgid (passwd:gid (job:user job)))
                     (setuid (passwd:uid (job:user job)))
                     (chdir (passwd:dir (job:user job)))
-- 
2.23.0

